File: dotnet/Pyrolite/Pickle/Objects/SetConstructor.cs
/* part of Pyrolite, by Irmen de Jong (irmen@razorvine.net) */

using System.Collections;
using System.Collections.Generic;

namespace Razorvine.Pickle.Objects
{

/// <summary>
/// This object constructor creates sets. (HashSet<object>)
/// </summary>
class SetConstructor : IObjectConstructor {

	public SetConstructor() {
	}

	public object construct(object[] args) {
		// create a HashSet, args=arraylist of stuff to put in it
		ArrayList elements=(ArrayList)args[0];
		IEnumerable<object> array=elements.ToArray();
		return new HashSet<object>(array);
	}
}

}

File: java/src/main/java/net/razorvine/pickle/objects/SetConstructor.java
package net.razorvine.pickle.objects;

import java.util.ArrayList;
import java.util.HashSet;

import net.razorvine.pickle.IObjectConstructor;

/**
 * This object constructor creates sets.
 * 
 * @author Irmen de Jong (irmen@razorvine.net)
 */
public class SetConstructor implements IObjectConstructor {

	public SetConstructor() {
	}

	public Object construct(Object[] args) {
		// create a HashSet, args=arraylist of stuff to put in it
		@SuppressWarnings("unchecked")
		ArrayList<Object> data = (ArrayList<Object>) args[0];
		return new HashSet<Object>(data);
	}
}
